### DISCUSSION TITLE
How to locate a module (API) in the source code to gain better understanding of DeepMD

### DISCUSSION THREAD

**Discussion by DingChangjie** [deepmodeling/deepmd-kit#788] [Q&A]
Dear developers and users : 
I'm trying to infer a frame from an already trained .pb file (to get its energy) by Python. There is , indeed, an example of [this](https://deepmd.readthedocs.io/en/master/use-deepmd-kit.html#model-inference), which prompted a module called deepmd.DeepPot should be used. However, I find the example a little ambiguous, especially for the meaning of "atype(atom type)" and "cell". 
I think it would be better for users to check the source code. I downloaded the v1.3.3 code (which I've been using), but find it quite troublesome for me to begin. Could anyone please share some experience on how to locate a module and understand its code ? 
For instance, is there any detailed documentation about every module ? Should I find the code from the freshly downloaded files, or the program built by cmake?
Thanks a lot in advance ! 
PS : I have some experience on python, and a little on C++ and cmake. However, till now I haven't had any experience on developing or maintaining a project ...

**Comment by njzjz** [comment]
The documentation will be added in next major version. You need to switch to `devel` branch to preview it.

**Reply by DingChangjie** [reply]
Thank you very much. I think I've found the module under deepmd-kit/deepmd/infer of `devel` branch
